synapse/util/retryutils.py

```diff
@@ -265,4 +265,6 @@ async def store_retry_timings() -> None:
                 logger.exception("Failed to store destination_retry_timings")
 
         # we deliberately do this in the background.
-        synapse.logging.context.run_in_background(store_retry_timings)
+        run_as_background_process(
+            f"store_retry_timings-{self.destination}", store_retry_timings
+        )
```

Comment:
The difference between `run_in_background` and `run_as_background_process` is that the former inherits the current logcontext, while the latter creates an entirely new one with separate metrics.